### ISSUE TITLE
siren on service call not working on Arlo Ultra

### ISSUE THREAD

**Issue by chris-chensy** [twrecked/hass-aarlo#235]
Hello,

I cannot activate the Arlo siren using the service call. When I use siren_on with the info below, it does nothing. More importantly, I cannot control the individual cameras. I have one base station for the three cameras. They do not appear individually on my list. When I use the switch, I can control my cameras individually. Is it because of my setting?

**Service Data**
entity_id: alarm_control_panel.aarlo_front_base
volume: 6
duration: 30

With many thanks,
Chris

**Comment by twrecked** [comment]
What kind of base station do you have?

Which version of Arlo are you using?

Can you send your `aarlo` switch configuration?

**Comment by chris-chensy** [comment]
> What kind of base station do you have?
> 
> Which version of Arlo are you using?
> 
> Can you send your `aarlo` switch configuration?

It is the base station that comes with the Arlo Ultra with the reference VMB5000r4. The camera model is VMC5040.

It is an Arlo Ultra that has been reported in another thread (spotlight issue). I have started a new one as I believe they are a separate problem with the spotlight.

My configuration of the switch is following:

switch:
  - platform: aarlo
    siren: True
    all_sirens: True
    snapshot: True
    siren_volume: 1
    siren_duration: 10

Cheers,
Chris

**Comment by sebr** [comment]
I'm happy to take a look at this in the next few days/week.

@chris-chensy - can I get some clarification on the issue please? Is this correct?
* Arlo Ultra siren works when triggered using the `switch.aarlo_camera` entity?
* Arlo Ultra siren does not work when triggered using the `aarlo.siren_on` service with `entity_id: alarm_control_panel.aarlo_hub`

**Comment by chris-chensy** [comment]
> I'm happy to take a look at this in the next few days/week.
> 
> @chris-chensy - can I get some clarification on the issue please? Is this correct?
> 
> * Arlo Ultra siren works when triggered using the `switch.aarlo_camera` entity?
> * Arlo Ultra siren does not work when triggered using the `aarlo.siren_on` service with `entity_id: alarm_control_panel.aarlo_hub`

Correct. 

I successfully call the service switch.turn_on for the siren of individual camera. 

But I cannot use the service call from aarlo domain using siren_on with my hub’s entity_id. Also, it seems lack of control for an individual camera if I have multiple cameras associated with the same hub in the aarlo domain.

Cheers
Chris

**Comment by twrecked** [comment]
Fixed by @sebr with #238 

I'll create a new release later today.

**Comment by twrecked** [comment]
I just created [0.6.8.15](https://github.com/twrecked/hass-aarlo/releases/tag/v0.6.18.15). It should be fixed in there.

**Comment by chris-chensy** [comment]
It seems there might be bugs with the built .15. This morning, my arlo baby temperature, motion states stopped updating. Restarting HA did not help. I had about 12-hour period with a flat temperature curve. I have to roll back to the build .13 to see the updates again.

**Comment by twrecked** [comment]
Were there any TraceBack in the logs?

**Comment by chris-chensy** [comment]
> Were there any TraceBack in the logs?

No specific issue found in terms of events. I found the following errors popped out quite regularly.
2020-05-02 21:45:35 WARNING (ArloEventStream) [pyaarlo] logged out? did you log in from elsewhere?

_2020-05-02 21:30:25 WARNING (ArloBackgroundWorker) [pyaarlo] error in response={'data': {'error': '2503', 'message': 'The device failed to connect. Try again.', 'reason': 'Camera is not streaming'}, 'success': False}

2020-05-02 21:25:20 WARNING (MainThread) [homeassistant.util.async_] Detected I/O inside the event loop. This is causing stability issues. Please report issue to the custom component author for aarlo doing I/O at custom_components/aarlo/pyaarlo/backend.py, line 77: r = self._session.post(url, json=params, headers=headers, timeout=timeout)_

**Comment by twrecked** [comment]
The first error means you had another login to Arlo using the same account. That will affect event updates.

The second one is an Arlo thing.

The third one is new. What version of home-assistant are you on? And what is your `aarlo:` configuration?

**Comment by chris-chensy** [comment]
> The first error means you had another login to Arlo using the same account. That will affect event updates.
**I use a dedicated account for HA which should not have another login…**
> 
> The second one is an Arlo thing.
**OK…**
> 
> The third one is new. What version of home-assistant are you on? And what is your `aarlo:` configuration?
**I’m using HA 0.109, which was updated two days ago. The config is the exact same as the wiki one, replacing the email and password. **

**Comment by twrecked** [comment]
My bad, sorry, the first one could also be Arlo. What frequency do they occur?

.15 is working for me but I'm on an older home assistant, I'll update it and take a look.

**Comment by sebr** [comment]
Yep - this is a [new feature in HA 109](https://www.home-assistant.io/blog/#detection-of-blocking-io-in-the-event-loop)

**Comment by sebr** [comment]
ps might I suggest closing this issue and opening a new one for the other issues?

**Comment by twrecked** [comment]
I created #243 for the I/O issue. 

Try adding the following to your config to help with any missing events/logout issue:

```yaml
aarlo:
  refresh_devices_every: 1
  stream_timeout: 125
```

As for why .15 didn't work, I have no idea. The only difference between the .13 and .15 was the siren code - you can see [here](https://github.com/twrecked/hass-aarlo/compare/v0.6.18.13...v0.6.18.15). Can you try .15 again with debug enabled? See if there are any exceptions.

**Comment by chris-chensy** [comment]
> I created #243 for the I/O issue.
> 
> Try adding the following to your config to help with any missing events/logout issue:
> 
> ```yaml
> aarlo:
>   refresh_devices_every: 1
>   stream_timeout: 125
> ```
> 
> As for why .15 didn't work, I have no idea. The only difference between the .13 and .15 was the siren code - you can see [here](https://github.com/twrecked/hass-aarlo/compare/v0.6.18.13...v0.6.18.15). Can you try .15 again with debug enabled? See if there are any exceptions.

Yes, I will try the new parameters in the config. 

Today, I met the same problem with the sensors paused updating for a couple of hours in the build .13. I have updated to .15 and see how it goes. It seems the problem happens when some criteria met. I will paste the debug info if I see it again.

Thanks for the efforts.
